### DISCUSSION TITLE
Save As ows

### DISCUSSION THREAD

**Discussion by drvshavva** [biolab/orange3#5682] [Q&A] [answered]
Hi, I am wondering how ows file saves all the parameters. Is there any documentation about save as ows.  

Is there any different usage from these two files to save as .ows such as parameters, library or class.
**orange3/orange3venv/lib/python3.7/site-packages/orangecanvas/application/canvasmain.py
orange3/orange3venv/lib/python3.7/site-packages/orangecanvas/scheme/readwrite.py**

Thank you for responding.

**Comment by janezd** [comment, marked as answer]
No, there's no such documentation, there won't be any and I wouldn't recommend trying to use these files for anything.

You can open the file (it's just an XML) and see for yourself what's in there. Every widget saves its internal data;  some widgets store it as literals (we plan to migrate all widgets to that format eventually), while for others the data is base64 encoded pickle. You can `base64decode` such text and give it to `pickle.loads`. But as you'll see, it essentially includes widget's internal variables, so the format can change at any time.

**Reply by drvshavva** [reply]
thank yo for quick response :smile:
